realm/realm-library/src/main/java/io/realm/mongodb/sync/SubscriptionSet.java

```diff
@@ -0,0 +1,231 @@
+     *
+     * @param name name of subscription to search for.
+     * @return the matching subscription or {@code null} if no subscription with that name was found.
+     */
+    @Nullable
+    Subscription find(String name);
+
+    /**
+     * Returns the current state of the SubscriptionSet. See {@link SubscriptionSet.State} for more
+     * details about each state.
+     *
+     * @return current state of the SubscriptionSet.
+     */
+    State getState();
+
+    /**
+     * Returns how many subscriptions are curently in this subscription set.
+     *
+     * @return the number of of subscriptions in the subscription set.
+     */
+    int size();
+
+    /**
+     * If {@link #getState()} returns {@link State#ERROR}, this method will return the reason.
+     * Errors can be fixed by modifying the subscription accordingly and then calling
+     * {@link #waitForSynchronization()}.
+     *
+     * @return the underlying error if the subscription set is in the {@link State#ERROR} state. For
+     * all other states {@code null} will be returned.
+     */
+    @Nullable
+    String getErrorMessage();
+
+    /**
+     * Wait for the subscription set to synchronize with the server. It will return when the
+     * server either accepts the set of queries and have downloaded data for them, or if an
+     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
+     * or wait for change listeners to trigger to see the downloaded data.
+     *
+     * If an error occurred, the underlying reason can be found through {@link #getErrorMessage()}.
+     *
+     * @return {@code true} if all current subscriptions was accepted by the server and data has
+     * been downloaded, or {@code false} if an error occurred.
+     */
+    boolean waitForSynchronization();
+
+    /**
+     * Wait for the subscription set to synchronize with the server. It will return when the
+     * server either accepts the set of queries and have downloaded data for them, or if an
+     * error has occurred. Note, that you will either need to manually call {@link Realm#refresh()}
+     * or wait for change listeners to trigger to see the downloaded data.
+     *
+     * If an error occurred, the underlying reason can be found through {@link #getErrorMessage()}.
+     *
+     * @param timeOut how long to wait for the synchronization to either succeed or fail.
+     * @param unit unit of time used for the timeout.
+     * @return {@code true} if all current subscriptions was accepted by the server and data has
+     * been downloaded, or {@code false} if an error occurred.
+     * @throws RuntimeException if the timeout is exceeded.
+     */
+    boolean waitForSynchronization(Long timeOut, TimeUnit unit);
+
+    /**
+     * Asynchronously wait for the subscription set to synchronize with the server. It will return when the
+     * server either accepts the set of queries and have downloaded data for them, or if an
```

Comment:
Also in multiple places:
```suggestion
     * server either accepts the set of queries and has downloaded data for them, or if an
```